client/rest-high-level/src/test/java/org/elasticsearch/client/DeleteResponseTests.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client;
+
+import org.elasticsearch.action.DocWriteResponse.Result;
+import org.elasticsearch.common.lucene.uid.Versions;
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.seqno.SequenceNumbers;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.Collections;
+
+public class DeleteResponseTests extends ESTestCase {
```

Comment:
testing these new classes requires to repeat a lot of similar checks, that would be avoided with a test class like AbstractXContentTestCase. But this new class does not implement toXContent hence it's hard to do that. I have faced something similar in #33176 and added a possible approach. The bigger problem though is that we need to make sure that all the response parser support unknown fields (for forward compatibility) and test that, which is something that is very easy to forget if we extend ESTestCase. @hub-cap said he needs to take some time to think about this testing problem caused by using different objects in the client.

Reply:
> #33176

Thanks for linking.

> The bigger problem though is that we need to make sure that all the response parser support unknown fields

Indeed I actually explicitly disabled support for unknown fields, which is a mistake now that I read your comment.